src/data/usborder.py

```diff
@@ -470,10 +471,41 @@ def AddBorderSegments(mexicoDoc, canadaDoc, coordinateLists):
             new_piece.append(new_piece[0])
         elif begin_anti_meridian >= 0 and end_anti_meridian == -1:
           new_piece.append(ls[i])
-      del ls[begin_anti_meridian+1 : end_anti_meridian-1]
-      lineStrings.append(new_piece)
- 
 
+      del ls[begin_anti_meridian+1 : end_anti_meridian]
+      return new_piece
+  return None
+
+# Split polygons crossing anti-meridians.
+for ls in lineStrings:
+  if ls[0] != ls[-1]:
+    print 'NOT A RING!'
+  new_ls = FixAntiMeridianPolygon(ls)
+  if new_ls is not None:
+    lineStrings.append(new_ls)
+
+# Reverse rings if necessary.
+for k, ls in enumerate(lineStrings):
+  coords = []
+  for c in ls:
+    xy = c.split(',')
+    coords.append([float(xy[0]), float(xy[1])])
+  lr = LinearRing(coords)
+  if not lr.is_ccw:
+    print 'Reversing non-CCW ring'
+    r = list(reversed(ls))
+    lineStrings[k] = r
+
+# Cleanup the coordinates to be rounded correctly with 9 precision numbers
+# (ie about 0.1mm)
+for k, ls in enumerate(lineStrings):
+  new_ls = []
+  for c in ls:
+    xy = c.split(',')
+    new_ls.append('%.9f,%.9f,0' % (float(xy[0]), float(xy[1])))
```

Comment:
I wonder if there is a small risk of getting unvalid geometry by doing this.
May be, it's better to test the gemetry validity before writing to kml

Thank you
Kusay Jaare from RED Technologies